Lib/fontTools/varLib/__init__.py

```diff
@@ -870,29 +922,35 @@ def build(designspace, master_finder=lambda s:s, exclude=[], optimize=True):
 	assert 0 == model.mapping[ds.base_idx]
 
 	log.info("Building variations tables")
-	if 'MVAR' not in exclude:
-		_add_MVAR(vf, model, master_fonts, axisTags)
-	if 'HVAR' not in exclude:
-		_add_HVAR(vf, model, master_fonts, axisTags)
-	if 'VVAR' not in exclude and 'vmtx' in vf:
-		_add_VVAR(vf, model, master_fonts, axisTags)
-	if 'GDEF' not in exclude or 'GPOS' not in exclude:
-		_merge_OTL(vf, model, master_fonts, axisTags)
-	if 'gvar' not in exclude and 'glyf' in vf:
-		_add_gvar(vf, model, master_fonts, optimize=optimize)
-	if 'cvar' not in exclude and 'glyf' in vf:
-		_merge_TTHinting(vf, model, master_fonts)
-	if 'GSUB' not in exclude and ds.rules:
-		_add_GSUB_feature_variations(vf, ds.axes, ds.internal_axis_supports, ds.rules, ds.rulesProcessingLast)
-	if 'CFF2' not in exclude and ('CFF ' in vf or 'CFF2' in vf):
-		_add_CFF2(vf, model, master_fonts)
-		if "post" in vf:
-			# set 'post' to format 2 to keep the glyph names dropped from CFF2
-			post = vf["post"]
-			if post.formatType != 2.0:
-				post.formatType = 2.0
-				post.extraNames = []
-				post.mapping = {}
+	try:
+		if 'MVAR' not in exclude:
+			_add_MVAR(vf, model, master_fonts, axisTags)
+		if 'HVAR' not in exclude:
+			_add_HVAR(vf, model, master_fonts, axisTags)
+		if 'VVAR' not in exclude and 'vmtx' in vf:
+			_add_VVAR(vf, model, master_fonts, axisTags)
+		if 'GDEF' not in exclude or 'GPOS' not in exclude:
+			_merge_OTL(vf, model, master_fonts, axisTags)
+		if 'gvar' not in exclude and 'glyf' in vf:
+			_add_gvar(vf, model, master_fonts, optimize=optimize)
+		if 'cvar' not in exclude and 'glyf' in vf:
+			_merge_TTHinting(vf, model, master_fonts)
+		if 'GSUB' not in exclude and ds.rules:
+			_add_GSUB_feature_variations(
+				vf, ds.axes, ds.internal_axis_supports, ds.rules, ds.rulesProcessingLast
+			)
+		if 'CFF2' not in exclude and ('CFF ' in vf or 'CFF2' in vf):
+			_add_CFF2(vf, model, master_fonts)
+			if "post" in vf:
+				# set 'post' to format 2 to keep the glyph names dropped from CFF2
+				post = vf["post"]
+				if post.formatType != 2.0:
+					post.formatType = 2.0
+					post.extraNames = []
+					post.mapping = {}
+	except Exception as e:
+		e.args = (f"In Designspace '{ds_path}': {', '.join(str(a) for a in e.args)}",)
```

Comment:
i'm not sure this should be done in here. I prefer if the client of varLib.build takes care of including (or not) the designspace.path in the error message